common/src/primitives/amount.rs

```diff
@@ -56,16 +58,28 @@ fn remove_right_most_zeros_and_decimal_point(s: String) -> String {
 }
 
 impl Amount {
-    pub const MAX: Self = Self::from_atoms(IntType::MAX);
+    pub const MAX: Self = Self::from_atoms(UnsignedIntType::MAX);
 
-    pub const fn from_atoms(v: IntType) -> Self {
+    pub const fn from_atoms(v: UnsignedIntType) -> Self {
         Amount { val: v }
     }
 
-    pub fn into_atoms(&self) -> IntType {
+    pub fn into_atoms(&self) -> UnsignedIntType {
         self.val
     }
 
+    pub fn from_signed(amount: SignedAmount) -> Option<Self> {
```

Comment:
Should we add simple tests for this and `into_signed` methods?

Reply:
Will do.